staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -596,21 +609,33 @@ func (s *store) List(ctx context.Context, key, resourceVersion string, pred stor
 			growSlice(v, 2048, len(getResp.Kvs))
 		}
 
-		// take items from the response until the bucket is full, filtering as we go
-		for _, kv := range getResp.Kvs {
-			if paging && int64(v.Len()) >= pred.Limit {
-				hasMore = true
-				break
```

Comment:
Does paging still work?

Reply:
It should. This change should not impact paging.
[TestListContinuation](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go#L1181) is passing.

Reply:
I opened #88674 to add more test coverage in this area.